src/root-hits.jl

```diff
@@ -0,0 +1,71 @@
+"""
```

Comment:
Please do use underscores, not hyphens, it's way more common to uses underscores in the Julia community.

Reply:
Ok, I actually just checked some packages and base, because I was sceptical. The *most* common trend seems to be to have short file names of one word or join two words together like "fastmath.jl" in julia base. Almost all names follow that convention. You're right that, when two words cannot be joined in a readable way, they are separated by underscores.

Given this I'd name the main script: "RootHitFile.jl" and the test script "test_RootHitFile.jl", since files are usually named after the central type or function of that script (if there is a central type or function).

Reply:
Well, *my* convention is that we use lower case file names, use underscores where a file relates to a struct with camel-case, and use "test_..." for test code (one test code file per source file) to that if you have multiple files open in editor tabs you can quickly see what's package and what's test code. And I'd like to to have some consistency across the legend packages. :-)

Reply:
Ah ok. So "root_hits.jl" and "test_root_hits.jl" for example would be your preference?

Reply:
Yes, resp. "root_hist_file.jl" if the main type in the file is called RootHistFile.

Question, though - is that name really ideal? People will assume it's a ".root"-type file that contains hits. Shouldn't we have something with "MaGe" and "Dario" in the name of the type instead?

Reply:
Yeah, good point, though I'm having trouble coming up with an alternative that's clear and concise. The fact that it's in the TextIO package should help make it clear that it's not a true .root format. I guess the full description would be: `MaGeDarioRootHitFile`, but that's pretty ugly. You could go down to `DarioRootHitFile`, but that's kinda long, too. Another option would be to encase that script in a module called `Dario`, so you'd have `Dario.RootHitFile` ?

Reply:
I think a submodule would be overkill here. How about `MaGeDarioOutputFile` or so?

Reply:
It's not the only output file from Dario. You also get actual ".root" files as part of the output.

Reply:
Sure, but we're not reading those here, that will go into LegendUpROOTIO.jl :-)

Reply:
I think it's a toss up if `MaGeDarioOutputFile` or `RootHitFile` is clearer. So, on the basis that `RootHitFile` is shorter and looks nicer I'd pick that.

Reply:
Sorry, but I really don't think people will even know it's from MaGe if called `RootHitFile`. The name should be intuitive to new users. It is MaGe's dario format right? Which specific to MaGe, not a Geant4-thing in general?

Reply:
Ah, I have an idea. How about `DarioHitFile` ? Since it's the file type that Dario outputs that gives you information on individual hits as opposed to summary info on whole events. That name would tell you what it contains and where it comes from. In the docstring you'd say it's from MaGe and has file ending ".root.hits".

And I think you're right about it being specific to MaGe as opposed to being a general geant4 format, but I don't know for certain.

Reply:
Or the plural: `DarioHitsFile`?

Reply:
I should add that the other files Dario outputs only give info on the event as a whole, like R90 or the event number. So, the root.hits format is truly the "hits" data format of the Dario output.

Reply:
`DarioHitsFile` seems fine to me.